From e279ae3d262906e2529ffc0131dc99be03287918 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 6 May 2017 13:49:05 -0400
Subject: [PATCH] Memory leak in mat coder

Fix a memory leak in mat coder triggered by a special crafted file

bug-debian: https://bugs.debian.org/870013
origin: https://github.com/ImageMagick/ImageMagick/commit/437a35e57db5ec078f4a3ccbf71f941276e88430
(cherry picked from commit 437a35e57db5ec078f4a3ccbf71f941276e88430)
---
 coders/mat.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/coders/mat.c b/coders/mat.c
index eb9d33d73..4b5edfe75 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -883,6 +883,7 @@ static Image *ReadMATImage(const ImageInfo *image_info,ExceptionInfo *exception)
   /*
      Open image file.
    */
+  quantum_info=(QuantumInfo *) NULL;
   image = AcquireImage(image_info);
 
   status = OpenBlob(image_info, image, ReadBinaryBlobMode, exception);
@@ -1298,7 +1299,8 @@ done_reading:
   }
 
   RelinquishMagickMemory(BImgBuff);
-  quantum_info=DestroyQuantumInfo(quantum_info);
+  if (quantum_info != (QuantumInfo *) NULL)
+    quantum_info=DestroyQuantumInfo(quantum_info);
 END_OF_READING:
   clone_info=DestroyImageInfo(clone_info);
   CloseBlob(image);
